### DISCUSSION TITLE
How to capture packets from Android client

### DISCUSSION THREAD

**Discussion by CanvaChen** [Auties00/WhatsappWeb4j#384] [General]
I have added the charles certificate as a system certificate, and other apps can see the https packets as plain text, but whatsapp is still in unknow state. I wonder if I need a client certificate, if so, where is the certificate path in the apk, and what other tools and steps are needed?

**Comment by ys1231** [comment]
Has the research been done?

**Comment by Darker935** [comment]
I suggest you to capture the sent/received nodes directly from the APK 

This is much much much easier than decrypt ws connection
